packages/matter.js/src/MatterController.ts

```diff
@@ -662,22 +665,34 @@ export class MatterController {
         );
     }
 
-    getNextAvailableSessionId() {
+    async getNextAvailableSessionId() {
         return this.sessionManager.getNextAvailableSessionId();
     }
 
-    async createSecureSession(
-        sessionId: number,
-        fabric: Fabric | undefined,
-        peerNodeId: NodeId,
-        peerSessionId: number,
-        sharedSecret: ByteArray,
-        salt: ByteArray,
-        isInitiator: boolean,
-        isResumption: boolean,
-        idleRetransmissionTimeoutMs?: number,
-        activeRetransmissionTimeoutMs?: number,
-    ) {
+    async createSecureSession(args: {
+        sessionId: number;
+        fabric: Fabric | undefined;
+        peerNodeId: NodeId;
+        peerSessionId: number;
+        sharedSecret: ByteArray;
+        salt: ByteArray;
+        isInitiator: boolean;
+        isResumption: boolean;
+        idleRetransmissionTimeoutMs?: number;
+        activeRetransmissionTimeoutMs?: number;
+    }) {
+        const {
+            sessionId,
+            fabric,
+            peerNodeId,
+            peerSessionId,
+            sharedSecret,
+            salt,
+            isInitiator,
+            isResumption,
+            idleRetransmissionTimeoutMs,
+            activeRetransmissionTimeoutMs,
+        } = args;
```

Comment:
If I'm reading correctly, looks like you could just stick ...args into createSecureSession and avoid the de-/re-structuring?

Reply:
right, adjusted in two places